src/EditorFeatures/CSharpTest/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs

```diff
@@ -580,6 +592,116 @@ public async Task TestNuGetAndVsixAnalyzer_MultipleNuGetAnalyzersCollectivelyRep
+                nugetAnalyzers: ImmutableArray.Create(firstNugetAnalyzer, secondNugetAnalyzer),
+                expectedNugetAnalyzersExecuted: true,
+                vsixAnalyzers: ImmutableArray.Create(vsixAnalyzer),
+                expectedVsixAnalyzersExecuted: true,
+                nugetSuppressors: ImmutableArray<NuGetSuppressor>.Empty,
+                expectedNugetSuppressorsExecuted: false,
+                vsixSuppressors: ImmutableArray.Create(vsixSuppressor),
+                expectedVsixSuppressorsExecuted: true,
+                new[]
+                {
+                    (Diagnostic("A", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer)),
+                    (Diagnostic("B", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer)),
+                    (Diagnostic("C", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer))
+                });
+
+            // All without overlap, verify the following:
+            //   1) No duplicate diagnostics
+            //   2) Both NuGet and Vsix analyzers execute
+            //   3) Appropriate diagnostic filtering is done - Nuget suppressor suppresses VSIX analyzer.
+            await TestNuGetAndVsixAnalyzerCoreAsync(
+                nugetAnalyzers: ImmutableArray.Create(firstNugetAnalyzer),
+                expectedNugetAnalyzersExecuted: true,
+                vsixAnalyzers: ImmutableArray.Create(vsixAnalyzer),
+                expectedVsixAnalyzersExecuted: true,
+                nugetSuppressors: ImmutableArray.Create(nugetSuppressor),
+                expectedNugetSuppressorsExecuted: true,
+                vsixSuppressors: ImmutableArray<VsixSuppressor>.Empty,
+                expectedVsixSuppressorsExecuted: false,
+                new[]
+                {
+                    (Diagnostic("A", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer)),
+                });
+
+            // Suppressors with duplicate support for VsixAnalzer, but not 100% overlap. Verify the following:
+            //   1) No duplicate diagnostics
+            //   2) Both NuGet and Vsix analyzers execute
+            //   3) Both Nuget and Vsix suppressors execute
+            //   4) Appropriate diagnostic filtering is done - Nuget suppressor and Vsix suppressor together suppresses VSIX analyzer.
+            await TestNuGetAndVsixAnalyzerCoreAsync(
+                nugetAnalyzers: ImmutableArray.Create(firstNugetAnalyzer),
+                expectedNugetAnalyzersExecuted: true,
+                vsixAnalyzers: ImmutableArray.Create(vsixAnalyzer),
+                expectedVsixAnalyzersExecuted: true,
+                nugetSuppressors: ImmutableArray.Create(partialNugetSuppressor),
+                expectedNugetSuppressorsExecuted: true,
+                vsixSuppressors: ImmutableArray.Create(vsixSuppressor),
+                expectedVsixSuppressorsExecuted: true,
+                new[]
+                {
+                    (Diagnostic("A", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer)),
```

Comment:
Can we verify this is a suppressed diagnostic, i.e. `Diagnostic.IsSuppressed` is true?

Reply:
it seems to me the suppressed diagnostics don't show up in the list I specify. I'll check the `IsSuppressed` property to verify but I think it will be `false`

Reply:
You probably need to change the below line of code that invokes `DiagnosticProviderTestUtilities.GetAllDiagnosticsAsync` to explicitly pass in `includeSuppressedDiagnostics: true`

http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs,630

Reply:
That will break existing tests.  Do you want those to be updated or am I missing something?

Reply:
er nevermind. There are no suppressors in the other tests, they'll be fine.

Reply:
Tests fixed.